src/org/sosy_lab/java_smt/basicimpl/SolverVersionChecker.java

```diff
@@ -0,0 +1,49 @@
+package org.sosy_lab.java_smt.basicimpl;
+
+import org.sosy_lab.java_smt.api.SolverOptimizationException;
+import org.sosy_lab.java_smt.api.Solvers;
+
+/**
+ * Utility class for checking solver versions and compatibility.
+ */
+public final class SolverVersionChecker {
+    private SolverVersionChecker() {}
+    
+    /**
+     * Check if the given MathSAT5 version has known optimization bugs.
+     * @param version The solver version string
+     * @throws SolverOptimizationException if the version has known issues
+     */
+    public static void checkMathSATVersion(String version) {
+        if (version.startsWith("1.7.2") || version.startsWith("1.7.3")) {
```

Comment:
We avoid hard-coded versions, because this blocks users from simply replacing solver binaries and test on their own. We try to provide as much from the solver as possible and useful for the user, and do not create artificial boundaries.